### PR TITLE
Make a note editable

### PR DESCRIPTION
**Pull request by apsitos**
Closes #15 and #3 

## Purpose

When viewing an existing reminder, the user can click on an “Edit” button where they can modify the title, notes, or date of the reminder.

## Approach

If you make a typo or use bad grammar, you can now fix it! Though if you’re the only one seeing the reminder, maybe you don’t care. 

### Learning

We did a lot of research that wasn’t super helpful. We had to screenhero with Brenna twice to get this working because Ember is stupid.

We read about the [mutable](http://emberjs.com/api/classes/Ember.Templates.helpers.html#method_mut) property, but didn’t end up using it.
 
### Open Questions and Pre-Merge TODOs

- [x] When editing a note, there is a “Save” button that commits the changes to the server.
- [x] When the user hits the save button, they are returned to the original individual note path (ie: reminders/1).

### Test coverage 

Wrote two tests: one checks that the edit button re-routes to the edit page, the second tests that we can mutate the title field and save the new text.

### Follow-up tasks

- [Issue #9](https://github.com/turingschool-projects/1610-remember-7/issues/9)

#### After

### CODE REVIEW

File: app/routes/edit.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+export default Ember.Route.extend({
+
+  model() {
+    return this.get('store').findRecord('reminder', this.model.id);
+  },
+
+  action: {
+    saveReminder() {
+      this.transitionTo("/reminders")
```
Comment (martensonbj) [inline app/routes/edit.js]: Is this firing? You might need a `sendAction()` line in your component.

**Review by martensonbj — CHANGES_REQUESTED** [review]
Hey - question about the communication between your component and route situation - Also there are merge conflicts. REsolve those please and then repush and we'll dig back in :)

File: app/routes/edit.js (hunk 1 above)
Comment (JeffBuss) [reply app/routes/edit.js]: It isn't, and is completely unnecessary at this point. We had originally tried handling it in the route, but instead used a link in the edit-reminder component. Good eyes!

**Comment by JeffBuss** [comment]
@martensonbj requested changes complete!

File: tests/unit/controllers/edit-test.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { moduleFor, test } from 'ember-qunit';
```
Comment (martensonbj) [inline tests/unit/controllers/edit-test.js]: Get rid of this file yo!

File: tests/acceptance/reminder-list-test.js [hunk 3]
```diff
@@ -31,6 +31,14 @@ test('clicking on an individual item', function(assert) {
...
+  visit('/');
+
+  andThen(function() {
+    assert.equal(find('.spec-no-reminder-prompt').length, 1, 'should display a prompt to create a reminder when none are present')
+  })
+})
+
 test('clicking new note button', function(assert) {
   visit('/');
   click('button');
```
Comment (martensonbj) [inline tests/acceptance/reminder-list-test.js]: What button? Add a class here or at least a button with a particular text

File: tests/acceptance/reminder-list-test.js [hunk 4]
```diff
@@ -56,10 +64,22 @@ test('adding new notes', function(assert) {
   })
 })
 
-test('add new note prompt', function(assert) {
-  visit('/');
+test('editing notes', function(assert) {
```
Comment (martensonbj) [inline tests/acceptance/reminder-list-test.js:67]: Let's set a variable to whatever the title is first, and check that the edited title is different. Right now it just asserts that whatever you typed matches what you typed.

**Review by martensonbj — CHANGES_REQUESTED** [review]
Few more things! Thanks for the effort guys :)

**Comment by apsitos** [comment]
@martensonbj Changes made!

**Review by martensonbj — APPROVED** [review]

**Pull request merged**